File: Beecrowd/2-AD-HOC/1136-Bingo!-4/1136.py
def diference(n:int, nuns:set) -> bool:
    for current in nuns:
        if n+current in nuns or (n-current)*-1 in nuns:
            return True
    return False

def bingo(ball: int, size:int, nuns:set) -> bool:
    for i in range(0, ball+1):
        if not diference(i, nuns):
            return False
    return True



def main():
    while True:
        data = input().split(' ')
        ball = int(data[0])
        size = int(data[1])

        if ball == 0 and size == 0:break;

        nuns = set(list(map(int, input().split(' '))))

        result = 'N'

        if bingo(ball, size, nuns):
            result = 'Y'
        print(result)
        
main()

File: Beecrowd/2-AD-HOC/1136-Bingo!-4/1136.cpp
#include <iostream>
#include <set>
using namespace std;

bool diference(int n, set<int>nuns){
    for (int current : nuns){
        //equcao modular do primeiro grau kkkk
        if(nuns.find(n+current) != nuns.end() | nuns.find((n-current)*-1) != nuns.end()){
            return true;
        }
    }
    return false;
}

bool bingo(int ball,int size, set<int>nuns){

    for(int i = 0; i<ball+1; i++){
        if (diference(i, nuns)==false){
            return false;
        }
    }
    return true;
    

}


int main(){
    
    int ball, help, size;
    char result;

    while (1){

        cin>>ball>>size;

        if (ball ==0 && size == 0)break;

        set<int>nuns = {};

        for(int i = 0; i<size; i++){
            cin>>help;
            nuns.insert(help);
        }

        result = 'N';
        if (bingo(ball, size, nuns)){
            result = 'Y';
        }

        cout<<result<<"\n";

        

    }

    return 0;
}